SatisfactorySaveParser/SatisfactorySave.cs

```diff
@@ -46,61 +46,68 @@ public SatisfactorySave(string file)
+                                var summary = reader.ReadChunkInfo();
 
-                            var startPosition = stream.Position;
-                            using (var zStream = new ZlibStream(stream, CompressionMode.Decompress, true))
-                            {
-                                zStream.CopyTo(buffer);
-                            }
+                                var subChunk = reader.ReadChunkInfo();
+                                Trace.Assert(subChunk.UncompressedSize == summary.UncompressedSize);
 
-                            // ZlibStream appears to read more bytes than it uses (because of buffering probably) so we need to manually fix the input stream position
-                            stream.Position = startPosition + subChunk.CompressedSize;
+                                var startPosition = stream.Position;
+                                using (var zStream = new ZlibStream(stream, CompressionMode.Decompress, true))
+                                {
+                                    zStream.CopyTo(buffer);
+                                }
 
-                            uncompressedSize += subChunk.UncompressedSize;
-                        }
+                                // ZlibStream appears to read more bytes than it uses (because of buffering probably) so we need to manually fix the input stream position
+                                stream.Position = startPosition + subChunk.CompressedSize;
 
+                                uncompressedSize += subChunk.UncompressedSize;
+                            }
 
-                        buffer.Position = 0;
+
+                            buffer.Position = 0;
 
 #if DEBUG
-                        //File.WriteAllBytes(Path.Combine(Path.GetDirectoryName(file), Path.GetFileNameWithoutExtension(file) + ".bin"), buffer.ToArray());
+                            //File.WriteAllBytes(Path.Combine(Path.GetDirectoryName(file), Path.GetFileNameWithoutExtension(file) + ".bin"), buffer.ToArray());
 #endif
 
 
-                        using (var bufferReader = new BinaryReader(buffer))
-                        {
-                            var dataLength = bufferReader.ReadInt32();
-                            Trace.Assert(uncompressedSize == dataLength + 4);
+                            using (var bufferReader = new BinaryReader(buffer))
+                            {
+                                var dataLength = bufferReader.ReadInt32();
+                                Trace.Assert(uncompressedSize == dataLength + 4);
 
-                            LoadData(bufferReader);
+                                LoadData(bufferReader);
+                            }
                         }
                     }
                 }
             }
+            catch (Exception ex) {
+                log.Error($"Error occurred in opening save file: {ex.Message}\n{ex.StackTrace}");
+                throw new FileNotFoundException();
```

Comment:
Don't randomly throw an exception, there's no guarantee this was actually caused by a FileNotFoundException. There's not much point in logging an error and rethrowing, because it'll either get handled or logged at a higher level

Reply:
Thanks. I completely glossed over the fact that more than just file not found exceptions could occur here for some reason.